edx_solutions_api_integration/courses/views.py

```diff
@@ -2694,29 +2694,60 @@ class OoyalaToBcoveConversion(MobileAPIView, IsStaffView):
     Controls a background task to convert the Ooyala Xblock's
     instances in given courses to Brightcove
     """
+    batch_size = 100
+
     def post(self, request):
         course_ids = request.data.get('course_ids')
         staff_user_id = request.data.get('staff_user_id')
         company_name = request.data.get('company_name')
         revert = str2bool(request.data.get('revert'))
+        run_on_all = str2bool(request.data.get('run_on_all'))
+        exclude_course_ids = request.data.get('exclude_course_ids', [])
+        email_ids = request.data.get('email_ids')
+        task_ids = []
 
-        if None in (course_ids, staff_user_id):
+        if None in (course_ids or run_on_all, staff_user_id):
```

Comment:
Would be better if we move check `not isinstance(exclude_course_ids, list)` here.
`if None in (course_ids or run_on_all, staff_user_id) or not isinstance(exclude_course_ids, list)`